internal/bgp/community/community.go

```diff
@@ -0,0 +1,141 @@
+	"strconv"
+	"strings"
+)
+
+// bgpCommunityType holds the type of the BGP community. Values currently supported are legacyBGPCommunity and
+// largeBGPCommunity. In the future, this can be expanded with "extendedBGPCommunity".
+type bgpCommunityType uint
+
+var (
+	ErrInvalidaCommunityValue = errors.New("invalid community value")
+	ErrInvalidCommunityFormat = errors.New("invalid community format")
+)
+
+const (
+	legacyBGPCommunity bgpCommunityType = iota
+	largeBGPCommunity
+)
+
+// largeBGPCommunityMarker is the prefix that shall be used to indicate that a given community value is of type large
+// community. If and when extended BGP communities will be supported, the largeBGPCommunityMarker allows us to
+// distinguish between extended and large communities.
+const largeBGPCommunityMarker = "large"
+
+// BGPCommunity holds the internal representation of a BGP community. It currently supports legacy and large BGP
+// communities with plans to expand this to extended communities in the future. A BGPCommunity object is intended to be
+// read only and as such all its methods directly operate on the type itself as opposed to using pointer receivers.
+type BGPCommunity struct {
+	communityType       bgpCommunityType
+	globalAdministrator uint32
+	localDataPart1      uint32
+	localDataPart2      uint32
+}
+
+// String returns the string representation of this community. Legacy communities will be printed in new-format,
+// "<AS number>:<community value>". Large communities will be printed as
+// ("<global administrator>:<localdata part 1>:<localdata part 2>").
+func (b BGPCommunity) String() string {
+	if b.communityType == legacyBGPCommunity {
+		return legacyCommunityToString(b.globalAdministrator)
+	}
+	return fmt.Sprintf("%d:%d:%d", b.globalAdministrator, b.localDataPart1, b.localDataPart2)
+}
+
+func legacyCommunityToString(c uint32) string {
+	upperVal := c >> 16
+	lowerVal := c & 0xFFFF
+	return fmt.Sprintf("%d:%d", upperVal, lowerVal)
+}
+
+// LessThan makes 2 different BGPCommunity objects comparable. For the sake of comparison, legacy communities are
+// considered to be large communities in format <legacy community>:0:0 and thus can be compared with large communities.
+func (b BGPCommunity) LessThan(c BGPCommunity) bool {
+	return b.globalAdministrator < c.globalAdministrator ||
+		(b.globalAdministrator == c.globalAdministrator && (b.localDataPart1 < c.localDataPart1 ||
+			(b.localDataPart1 == c.localDataPart1 && b.localDataPart2 < c.localDataPart2)))
+}
+
+// IsLegacy returns true if this community is of type legacy.
+func (b BGPCommunity) IsLegacy() bool {
+	return b.communityType == legacyBGPCommunity
+}
+
+// IsLarge returns true if this community is of type large.
+func (b BGPCommunity) IsLarge() bool {
+	return b.communityType == largeBGPCommunity
+}
+
+// ToLegacy returns the uint32 representation of this community. If this community is not of type legacy, throw an
+// error.
+func (b BGPCommunity) ToLegacy() (uint32, error) {
```

Comment:
I'd call this ToUint32, as it feels counterintuitive to say "I convert a legacy community to legacy". Hope it makes sense

Reply:
I removed it all together with the interface implementation